flink-runtime/src/main/java/org/apache/flink/runtime/highavailability/AbstractHaServices.java

```diff
@@ -247,7 +247,7 @@ private LeaderElection createLeaderElection(String leaderName) throws Exception
         leaderElectionService.startLeaderElectionBackend();
 
         closeableRegistry.registerCloseable(leaderElectionService);
-        return leaderElectionService.createLeaderElection();
+        return leaderElectionService.createLeaderElection(leaderName);
```

Comment:
hmm. This isn't really a contender Id (which would identify a single contender), but rather the contender _type/role_ isn't it? These are values like "restserver", "dispatcher" etc..

Like, you can have 2 jobmanager _contenders_ contending for the _role_ of the leading jobmanager.

Reply:
Isn't this actually a bug? How does this work with the multi component LES?

Reply:
phew, you got me on that one. :-D I was confused for a second as well. It's indeed a bit confusing right now. The `contenderID` isn't passed forward to the `LeaderElectionDriver` within the `DefaultLeaderElectionService` right now. It becomes used as soon as we're switching from the `LeaderElectionDriver` interface to `MultipleComponentLeaderElectionDriver` within `DefaultLeaderElectionService`. That's going to happen in PR #22661

I'm going to add comments to explain this.

Reply:
Is that still missing from the PR?

Because it looks like we're always passing a constant value into `AbstractHaServices#createLeaderElection`, that forwards that same static value into the LeaderElection and DriverFactory.

Reply:
The `contenderID` that's passed in through `createLeaderElection` is saved in the `DefaultLeaderElection` and forwarded in each call to the `DefaultLeaderElectionService`. But we don't forward it for now to the driver. The driver gets it through the `MultipleComponentLeaderElectionDriverAdapter` for now (which gets the value through [AbstractHaServices:246](https://github.com/XComp/flink/blob/f4c8cdb8f370de26f55759e7ea4b6f5461c7a806/flink-runtime/src/main/java/org/apache/flink/runtime/highavailability/AbstractHaServices.java#L246)).

In FLINK-31783 (PR #22661), we will remove the adapter functionality and integrate the `MultipleComponentLeaderElectionDriver` into `DefaultLeaderElectionService`. I replaced the variable usage with a static string that should make it clearer that the value isn't used for now.